### PR TITLE
Added shortest common Supersequence in cpp

### PR DESCRIPTION
**Pull request by bitaashna**
I have added **'Shortest Common SuperSequence'** in **Cpp**
#167

### CODE REVIEW

File: Code/C++/print_shortest_supersequence.cpp [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+  Shortest Common Supersequence: agxgtxayb
+        Length of Shortest Common SuperSequence: 9
+  
+  Input 2:
+  Enter string s1: luck
+  Enter string s2: cyk
+
+  Output 2:
+  Shortest Common Supersequence: lucyk
+        Length of Shortest Common SuperSequence: 5
```
Comment (OjusWiZard) [inline Code/C++/print_shortest_supersequence.cpp:122]: Align the indentation as above

**Review by OjusWiZard — CHANGES_REQUESTED** [review]
Add the Algo name to the Readme too

**Comment by bitaashna** [comment]
@OjusWiZard I have made the changes..Please review!

**Review by OjusWiZard — APPROVED** [review]
LGTM

File: Code/C++/print_shortest_supersequence.cpp [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    //string for storing the shortest supersequence
+    string s;
+    //calculating length of string
+    int n = s1.length();
+    int m = s2.length();
+    //constructing an 2D matrix of size n+1 and m+1
+    int dp[n + 1][m + 1];
+
+    for (int i = 0; i <= n; i++)
+        for (int j = 0; j <= m; j++) { //base condition
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:28]: Add comments above the line

File: Code/C++/print_shortest_supersequence.cpp [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+            //initializing the first row and first column as 0
+            if (i == 0 || j == 0)
+                dp[i][j] = 0;
+        }
+
+    //finding LCS
+    for (int i = 1; i < n + 1; i++) {
+        for (int j = 1; j < m + 1; j++) { //cheking if character matches or not
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:36]: Add comments above the line

File: Code/C++/print_shortest_supersequence.cpp [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+  
+  Input 3:
+  Enter string s1: acbcf
+  Enter string s2: abcdaf
+
+  Output 3:
+  Shortest Common Supersequence: acdcdaf
+  Length of Shortest Common SuperSequence: 7
+  
+  * For s1 = "a c b c f" and s2 = "a b c d a f"
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:132]: Mention "Explanation" here

File: Code/C++/print_shortest_supersequence.cpp [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+  * and LCS matrix for this is :
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:133]: Remove * from the front throughout from 132 - 172.

File: Code/C++/print_shortest_supersequence.cpp [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+  *  matrix[c][a] = 3   : Result[a f]
+  *  matrix[c][d] = 3   : Result[d a f]
+  *  matrix[c][c] = 3   : Result[c d a f]
+  *  matrix[b][b] = 2   : Result[b c d a f]
+  *  matrix[c][a] = 1   : Result[c b c d a f]
+  *  matrix[a][a] = 1   : Result[a c b c d a f]  ===> this is our ShortestCommonSuperSequence.
+  *
+         
+
+* Time Complexity: O(m*n)
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:171]: Specify m and n

File: Code/C++/print_shortest_supersequence.cpp (hunk 5 above)
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:133]: Remove * from the front throughout from 132 - 172.

File: Code/C++/print_shortest_supersequence.cpp (hunk 4 above)
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:132]: Mention "Explanation" here

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by bitaashna** [comment]
@raksha009 I have made the changes.Please review.

File: Code/C++/print_shortest_supersequence.cpp [hunk 7]
```diff
@@ -0,0 +1,175 @@
...
+    for (int i = 0; i <= n; i++)
+        for (int j = 0; j <= m; j++) { 
+            
+            if (i == 0 || j == 0)
+                dp[i][j] = 0;
+        }
+
+    //finding LCS
+    for (int i = 1; i < n + 1; i++) {
+        for (int j = 1; j < m + 1; j++) { //cheking if character matches or not
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:37]: Hey, plz add comments above the line throughout yr code.

File: Code/C++/print_shortest_supersequence.cpp [hunk 8]
```diff
@@ -0,0 +1,175 @@
...
+            i--;
+            j--;
+        }
+        //if current character in s1 and s2 are different
+        else if (dp[i][j - 1] > dp[i - 1][j]) { //push current character of s2 in result
+            s.push_back(s2[j - 1]);
+            //reduce value of j
+            j--;
+        }
+        else { //push character of s1 in result
```
Comment (raksha009) [inline Code/C++/print_shortest_supersequence.cpp:68]: add comments above the line

**Review by raksha009 — CHANGES_REQUESTED** [review]

**Comment by bitaashna** [comment]
@raksha009 Done.Please review!

**Review by raksha009 — APPROVED** [review]

**Pull request merged**